tests/foreman/cli/test_reporttemplates.py

```diff
@@ -349,3 +349,135 @@ def test_negative_delete_locked_report(self):
+
+        :id: a4b777db-142e-4871-a42e-e93887464986
+
+        :setup: User with reporting access rights, some host with applied errata
+
+        :steps:
+
+            1. hammer report-template schedule --wait ...
+
+        :expectedresults: A report is generated asynchronously
+
+        :CaseImportance: Medium
+        """
+
+    @tier2
+    @stubbed()
+    def test_positive_generate_nonblocking_download(self):
+        """Generate an Applied Errata report
+
+        :id: a4b777db-142e-4871-a42e-e93887464986
+
+        :setup: User with reporting access rights, some host with applied errata
+
+        :steps:
+
+            1. hammer report-template schedule ...
+            2. hammer report-template report-data --job-id= ...
+
+        :expectedresults: A report is generated asynchronously
+
+        :CaseImportance: Medium
+        """
+
+    @tier2
+    @stubbed()
+    def test_positive_generate_email_compressed(self):
+        """Generate an Applied Errata report, get it by e-mail, compressed
+
+        :id: a4b877db-143e-4871-a42e-e93887464986
+
+        :setup: User with reporting access rights, some host with applied errata
+
+        :steps:
+
+            1. hammer report-template schedule ...
+
+        :expectedresults: A report is generated asynchronously, the result
+                          is compressed and mailed to the specified address
+
+        :CaseImportance: Medium
+        """
+
+    @tier2
+    @stubbed()
+    def test_positive_generate_email_uncompressed(self):
+        """Generate an Applied Errata report, get it by e-mail, uncompressed
+
+        :id: a4b977db-143f-4871-a42e-e93887464986
+
+        :setup: User with reporting access rights, some host with applied errata
+
+        :steps:
+
+            1. hammer report-template schedule ...
+
+        :expectedresults: A report is generated asynchronously, the result
+                          is not compressed and is mailed
+                          to the specified address
+
+        :CaseImportance: Medium
+        """
+
+    @tier2
+    @stubbed()
+    def test_negative_bad_email(self):
+        """ Report can't be generated when incorrectly formed mail specified
+
+        :id: a4ba77db-144e-4871-a42e-e93887464986
+
+        :setup: User with reporting access rights, some host with applied errata
+
+        :steps:
+
+            1. hammer report-template schedule ...
+
+        :expectedresults: Error message about wrong e-mail address, no task is triggered
+
+        :CaseImportance: Medium
+        """
+
+    @tier2
+    @stubbed()
+    def test_negative_nonauthor_of_report_cant_download_it(self):
+        """The resulting report should only be downloadable by
+           the user that generated it. Check.
+
+        :id: a4bc77db-146e-4871-a42e-e93887464986
+
+        :setup: Installed Satellite, user that can list running tasks
+
+        :steps:
+
+            1. hammer report-template schedule
+            2. hammer report-template report-data
```

Comment:
different user credentials need to be used here ofc, also make sure the user is not admin